Flaky test workflow failures

直近2件のPRで連続して、Checksの1回目が失敗し、再試行で成功した点が気になっています。どちらも`pnpm test`の異常終了ですが、原因は違うように見えます。

## https://github.com/vivliostyle/vivliostyle-cli/actions/runs/20420010345/attempts/1

```
Run pnpm test

...

 ✓ tests/server.test.ts (14 tests) 1072ms
   ✓ vite-plugin-dev-server > serves single document  302ms
/Users/runner/work/_temp/48c64368-8df0-49fb-bffc-662509dfa526.sh: line 1:  7582 Abort trap: 6           pnpm test
 ELIFECYCLE  Test failed. See above for more details.
Error: Process completed with exit code 134.
```

#### 推測

すでに`vite.config.ts`にコメントがある件は、元情報（[Common Errors - Vitest](https://vitest.dev/guide/common-errors.html#segfaults-and-native-code-errors)）では特にOSを名指ししていません。出ているエラーもまさにこれなので、MacやWindowsも`forks`とするべきかもしれません。

https://github.com/vivliostyle/vivliostyle-cli/blob/de76f1b484d544b8878e8b36b36c841d75f43d06/vite.config.ts#L20-L24

## https://github.com/vivliostyle/vivliostyle-cli/actions/runs/20448585698/attempts/1

```
Run pnpm test

...

 ✓ tests/pdf.test.ts (1 test) 12527ms
   ✓ pdf output > generate pdf without errors  12526ms
 ELIFECYCLE  Test failed. See above for more details.
Error: Process completed with exit code 1.
```

#### 推測

何も出ていないのでなんともいえませんが、テスト自体は成功しているのでクリーンアップ周りかもしれません。

--- a/.changeset/dirty-chicken-hear.md
+++ b/.changeset/dirty-chicken-hear.md
@@ -1,0 +1,5 @@
+---
+'@vivliostyle/cli': patch
+---
+
+Run tests with Vitest's default `forks` pool on all platforms to avoid native-module segfaults in CI.

--- a/vite.config.ts
+++ b/vite.config.ts
@@ -16,11 +16,5 @@ export default defineConfig({
       NO_COLOR: 'true',
     },
     clearMocks: true,
-    pool:
-      process.platform === 'linux'
-        ? // In Linux, we need to use forks mode to avoid segfaults
-          // https://vitest.dev/guide/common-errors.html#segfaults-and-native-code-errors
-          'forks'
-        : 'threads',
   },
 });

